webapp/javascript/pages/ExploreView.tsx

```diff
@@ -0,0 +1,309 @@
+          <button>Export</button>
+          <button>Single</button>
+          <button>Comparison</button>
+          <button>Diff</button>
+        </div>
+      </div>
+      <table className={styles.exploreTable}>
+        <thead>
+          <tr>
+            <th>{groupsData[0]?.tagName === '*' ? 'App' : 'Tag'} name</th>
+            <th>Event count</th>
+            <th>avg samples</th>
+            <th>std deviation samples</th>
+            <th>min samples</th>
+            <th>max samples</th>
+            <th>50% tbd</th>
+            <th>75% tbd</th>
+            <th>max tbd</th>
+            <th>cost tbd</th>
+          </tr>
+        </thead>
+        <tbody>
+          {groupsData.map(({ tagName, color, data }) => {
+            const mean = calculateMean(data.samples);
+
+            return (
+              <tr
+                className={
+                  tagName === groupByTagValue ? styles.activeTagRow : ''
+                }
+                onClick={() => handleGroupByTagValueChange(tagName)}
+                key={tagName}
+              >
+                <td>
+                  {tagName === '*' ? (
+                    appName
+                  ) : (
+                    <div className={styles.tagName}>
+                      <span
+                        className={styles.tagColor}
+                        style={{ backgroundColor: color?.toString() }}
+                      />
+                      {tagName}
+                    </div>
+                  )}
+                </td>
+                <td>{data.samples.length}</td>
+                <td>{mean.toFixed(2)}</td>
+                <td>{calculateStdDeviation(data.samples, mean).toFixed(2)}</td>
+                <td>{Math.min(...data.samples)}</td>
+                <td>{Math.max(...data.samples)}</td>
+                <td>50,987 mock</td>
+                <td>76,200 mock</td>
+                <td>100,000 mock</td>
+                <td>$ 250 / hr mock</td>
+              </tr>
+            );
+          })}
+        </tbody>
+      </table>
+    </>
+  );
+}
+
+function ExploreHeader({
+  appName,
+  tags,
+  selectedTag,
+  handleTagChange,
+}: {
+  appName: Maybe<string>;
+  tags: TagsState;
+  selectedTag: string;
+  handleTagChange: (value: string) => void;
+}) {
+  const tagKeys = Object.keys(tags.tags);
+
+  const handleClick = (e: ClickEvent) => {
+    handleTagChange(e.value);
+  };
+
+  return (
+    <div className={styles.header}>
+      <span className={styles.title}>
+        {appName.isJust ? appName.value : ''}
+      </span>
+      <div className={styles.query}>
+        <span className={styles.selectName}>grouped by</span>
+        <Dropdown
+          label="tags"
+          value={selectedTag ? `tag: ${selectedTag}` : 'select tag'}
+          onItemClick={handleClick}
+        >
+          {tagKeys.map((tagName) => (
+            <MenuItem key={tagName} value={tagName}>
+              {tagName}
+            </MenuItem>
+          ))}
```

Comment:
I already mentioned this, but let's handle the case where there's no tags available.